### PR TITLE
Clarify combine filter examples

### PR DESCRIPTION
**Pull request by agowa**
##### SUMMARY
Provide better examples for the ansible.builtin.combine filter.

Also add a note that keyword parameters need to be listed after positional ones.

##### ISSUE TYPE
- Docs Pull Request

+label: docsite_pr

### CODE REVIEW

File: lib/ansible/plugins/filter/combine.yml [hunk 1]
```diff
@@ -34,9 +34,16 @@ DOCUMENTATION:
...
 
     # ab => {'a':1, 'b':3, 'c': 4}
-    ab: {{ {'a':1, 'b':2} | combine({'b':3, 'c':4}) }}
+    ab: {{ {'a':1, 'b':2} | ansible.builtin.combine({'b':3, 'c':4}) }}
 
-    many: "{{ dict1 | combine(dict2, dict3, dict4) }}"
+    many: "{{ dict1 | ansible.builtin.combine(dict2, dict3, dict4) }}"
+    
+    # Nested merge, I.E. Defaults using complex data structures with partial customization overwrites
+    # Note: Keyword parameters need to be listed *after* positional parameters
```
Comment (mattclay) [inline lib/ansible/plugins/filter/combine.yml]: This shouldn't be explained in each plugin. Also, this is already covered in documentation:

https://docs.ansible.com/ansible/latest/plugins/filter.html#using-filter-plugins

> When passing both types, positional arguments should go first.

```suggestion
```

**Review by mattclay — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/filter/combine.yml (hunk 1 above)
Comment (agowa) [reply lib/ansible/plugins/filter/combine.yml]: Can we tweak the page generator to add this note to every filter plugin that uses both types?
Maybe where currently the text starting with "This describes keyword parameters of the filter. " is generated?

Comment (agowa) [reply lib/ansible/plugins/filter/combine.yml]: Moved this note to: https://github.com/ansible-community/antsibull-docs/pull/101

**Comment by agowa338** [comment]
!needs_revision
ready_for_review

**Review by mattclay — APPROVED** [review]

**Pull request merged**